include/boost/math/special_functions/lambert_w.hpp

```diff
@@ -1223,19 +1215,19 @@ T lambert_w_negative_rational_float(T z, const Policy& pol)
 }
 
 //! Lambert_w0 @b 'float' implementation, selected when T is 32-bit precision.
-template <class T, class Policy>
-inline T lambert_w0_imp(T z, const Policy& pol, const boost::integral_constant<int, 1>&)
+template <typename T, typename Policy>
+inline T lambert_w0_imp(T z, const Policy& pol, const std::integral_constant<int, 1>&)
 {
-  static const char* function = "boost::math::lambert_w0<%1%>"; // For error messages.
+  static std::string function = "boost::math::lambert_w0<%1%>"; // For error messages.
```

Comment:
What's the motivation here?  A string literal avoids the (tiny) runtime overhead of initializing a std::string which we don't use (as that type) anyway?

Reply:
Mostly to remove non-smart pointers that are floating around as part of the modernization campaign. If there is a `BOOST_CXX20_CONSTEXPR` macro it could become `constexpr`.

Reply:
I reverted it to `const char*`; I would say that is more idiomatic than a C style array.